controller/src/main/java/org/jboss/as/controller/AbstractOperationContext.java

```diff
@@ -707,6 +712,7 @@ public void rollback() {
                     persistenceResource.commit();
                 }
             }
+            publishCapabilityRegistry();
```

Comment:
There are 3 cases here:

1) persistenceResource == null
2) persistenceResource != null && resultAction == ResultAction.KEEP
3) persistenceResource != null && resultAction == ResultAction.ROLLBACK

The publishCapabilityRegistry(); should only happen in the first two cases.

The 3rd case already gets handled by the subsequent call to executeResultHandlerPhase, which calls operationRollingBack if resultAction != ResultAction.KEEP.